tests/integration/test_model_errors.py

```diff
@@ -43,6 +44,32 @@ def test_optimize_missing_subscript():
         model = Model(data_df, parsed_lines)
 
 
+def test_lines_in_wrong_order_for_primes():
+    data_df = pandas.read_csv(os.path.join(test_dir, "bernoulli.csv"))
+
+    model_string = """
+    mu ~ normal(-0.5, 0.3);
+    y' ~ bernoulli_logit(mu[group]);
+    """
+
+    with pytest.raises(Exception, match="The primed uses must come last"):
+        model = Model(data_df, model_string=model_string)
+
+
+def test_lines_in_wrong_order_for_assignments():
+    data_df = pandas.read_csv(os.path.join(test_dir, "bernoulli.csv"))
+
+    model_string = """
+    y' ~ bernoulli_logit(mu[group]);
+    mu[group]' = 0.1 + mu2[group]; # If you remove the '0.1 +' this fails to parse
```

Comment:
The compiler gives a type error if you remove the `0.1 +` part of this line. Not sure what is going on, but presumably it can't infer the type for mu2?

Reply:
I left out a type signature for assignment. Added in the new commit.